### PR TITLE
Add env variables for easy switch of local development and add setup instructions to readme

### PR DESCRIPTION
**Pull request by andreasRu**
This PR:
- adds env variables with fallback to the default values, that enables easier switching for local development with Lucee Express. That should motivate external contributors to enhance the code.
- adds a template for quick setup/update of the RESTful service mappings in local development with a simple call.
- adds an installation and setup guide for running the dev environment

That should help contributors setting up a deveopment fast.

Note: I did this with my best effort to not break existing code, but I did minor refactoring. I wasn't able to test it in a proper configuration as of production because I don't know all the internal web server and servlet container production settings.

### CODE REVIEW

**Comment by cfmitrah** [comment]
@andreasRu http://localhost:8888/download/updateRestMappings.cfm first request to this works and shows the message as **REST mappings updated for localhost**. But the second request throws errors like **key [S3ROOT] doesn't exist in the request scope** on UpdateProvider.cfc: line 3


key [S3ROOT] doesn't exist in the request scope
The Error Occurred in
D:\lucee-data-provider\update\UpdateProvider.cfc: line 3

    1: component restpath="/provider" rest="true" {
    2:
    3: request.s3Root = server.system.environment["LUCEE_DATA_PROVIDER_S3ROOT"] ?: request.s3Root; "s3:///lucee-downloads/";
    4: request.s3URL = server.system.environment["LUCEE_DATA_PROVIDER_S3URL"] ?: "https://s3-eu-west-1.amazonaws.com/lucee-downloads/";
    5: variables.cdnURL="https://cdn.lucee.org/";

called from D:\lucee-data-provider\update\UpdateProvider.cfc: line 1
called from D:\lucee-data-provider\download\updateRestMappings.cfm: line 19

File: download/updateRestMappings.cfm [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<cfscript>
+    /***
+    * Run this template to add/update Restful services from within your web-context (e.g. http://localhost:8888/download/updateRestMappings.cfm ) 
+    * that will add/update the Rest Service. After running, confirm with http://localhost:8888/rest (to activate rest listings go
+    * to: Lucee Administrator -> Archives & Resources -> Rest, and activate "List services")
+    ***/
+
+    if( cgi.SERVER_NAME=="localhost") {
+        basePath=getDirectoryFromPath( getTemplatePath() );
```
Comment (cfmitrah) [inline download/updateRestMappings.cfm:9]: getTemplatePath() is deprecated. https://docs.lucee.org/reference/functions/gettemplatepath.html

Comment (andreasRu) [reply download/updateRestMappings.cfm]: @cfmitrah Thanks for the help!!! I've changed the variable assignment to not get the variable from the request scope and also, got rid of the getTemplatePath().

Could you do another retry?

Comment (cfmitrah) [reply download/updateRestMappings.cfm]: @andreasRu I've made some tests on this PR. It works fine

**Comment by DominicWatson** [comment]
Hey @andreasRu - I've just jumped on this project due to some urgent need to move it off its existing server. This is a real shame that this never made it into the project.

However, the code has been somewhat refactored and most of this PR is now redundant, though the effort is greatly appreciated and certainly aspects of it that could be continued/revisited.

The current refactor has happened quite quickly, highlights:

1. Not relying on local disk for the most part - cached artifacts stored and fetched from S3
2. Structure rewored
3. Containerised
4. Github actions created to do test, build + deploy pipelines

Its all very rough, but a much better base to kick off from. I'm unfortunately going to have to close this PR but would welcome any further contributions if you were able.

Regards

**Pull request closed without merging**